rosidl_typesupport_cpp/include/rosidl_typesupport_cpp/type_support_map.h

```diff
@@ -22,15 +22,26 @@ extern "C"
 {
 #endif
 
+/// Contains all available C++ typesupport handles to choose from.
 typedef struct type_support_map_t
 {
   // TODO(dirk-thomas) const should not be defined for the fields
   // but should be set for the struct when it is being used
   // same for rosidl_message_type_support_t et al
+
+  /// Length of the `typesupport_identidentifier`, `symbol_name` and `data` arrays.
   const size_t size;
+
+  /// The ROS 2 package this is generated from.
   const char * package_name;
+
+  /// Array of identifiers for the type_supports.
   const char * const * typesupport_identifier;
+
+  /// Names of the symbols associated with each type support.
```

Comment:
Nitpick: to match the other members this could mention that it is an array, e.g.:

> Array of symbol names for the type_supports

Maybe also clarify what the symbol is being used for / what signature it has? Pointing to `ROSIDL_TYPESUPPORT_INTERFACE__MESSAGE_SYMBOL_NAME`?